app/gui2/src/components/documentation/DocsSynopsis.vue

```diff
@@ -0,0 +1,123 @@
+<script setup lang="ts">
+import iconImportant from '@/assets/icon-important.svg'
+import iconInfo from '@/assets/icon-info.svg'
+import type { Doc } from '@/util/docParser'
+
+const props = defineProps<{ sections: Doc.Section[] }>()
+</script>
+
+<template>
+  <!-- eslint-disable vue/no-v-html -->
+  <div v-if="props.sections.length > 0" class="sectionContent">
+    <div v-for="(section, i) in props.sections" :key="i">
+      <p v-if="'Paragraph' in section" class="paragraph">
+        <span v-html="section.Paragraph.body ?? 'Invalid body'"></span>
+      </p>
+      <p v-if="'Keyed' in section" class="paragraph">
+        {{ section.Keyed.key + ': ' + section.Keyed.body }}
+      </p>
+      <div
+        v-if="'Marked' in section"
+        :class="[
+          {
+            backgroundInfo: section.Marked.mark == 'Info',
+            backgroundImportant: section.Marked.mark == 'Important',
+          },
+          'markedContainer',
+        ]"
+      >
+        <div v-if="'header' in section.Marked" class="markedHeader">
+          <img :src="section.Marked.mark == 'Info' ? iconInfo : iconImportant" class="markedIcon" />
+          {{ section.Marked.header }}
+        </div>
+        <p class="paragraph" v-html="section.Marked.body ?? 'Invalid body'" />
+      </div>
+      <ul v-if="'List' in section">
+        <li
+          v-for="(item, index) in section.List.items"
+          :key="index"
+          v-html="item ?? 'Invalid item'"
+        ></li>
+      </ul>
+      <ul v-if="'Arguments' in section">
+        <li v-for="(arg, index) in section.Arguments.args" :key="index">
+          <span class="argument">{{ arg.name }}</span
+          >:&nbsp;
+          <span v-html="arg.description ?? 'Invalid description'"></span>
+        </li>
+      </ul>
+    </div>
+  </div>
+  <!-- eslint-enable vue/no-v-html -->
+</template>
+
+<style scoped>
+/* Common parts. */
+
+ul {
+  margin: 0;
+  padding: 0;
+  list-style-type: none;
+  list-style-position: inside;
+
+  li {
+    &:before {
+      content: '•';
+      font-size: 13px;
+      font-weight: 600;
+      margin-right: 6px;
+    }
+  }
+}
+
+.paragraph {
+  margin: 0;
+  padding: 1px 0;
+}
+
+.argument {
+  font-weight: 800;
+}
+
+/* Marked sections, such as `Info` and `Important` sections. */
+
+.backgroundInfo {
+  background-color: var(--enso-docs-info-background-color);
+}
+
+.backgroundImportant {
+  background-color: var(--enso-docs-important-background-color);
+}
+
+.markedContainer {
+  border-radius: 0.5rem;
+  padding: 0.5rem;
+  margin: 0.5rem 0;
+}
+
+.markedHeader {
+  font-weight: 600;
+  font-size: 13px;
+  margin: 0;
+  display: flex;
+  align-items: center;
+  gap: 0.25em;
+}
+
+.markedIcon {
+  pointer-events: none;
+  width: 1em;
+  height: 1em;
+  margin: 0 0.05em 0 0.05em;
+  vertical-align: -0.1em;
+  fill: none;
+}
+
+/* Code. The words emphasized with backticks. */
+
+:deep(code) {
```

Comment:
I believe we're going with `:is()` instead of `:deep()`? Pinging @Frizi for confirmation. Not sure whether there's a meaningful difference apart from `:is` being a relatively new feature. ([caniuse](https://caniuse.com/?search=is()) - Chromium 88+, Safari 14+, FF 78+)